### ISSUE TITLE
Unintuitive result with 1.4.0 types

### ISSUE THREAD

**Issue by calvinwyoung** [supercharge/promise-pool#20]
Hi there, thanks for putting this library together — it's a great lightweight solution to a common problem. I just played around with the latest 1.4.0 release, and noticed something somewhat unintuitive. Suppose I have the following snippet:

```typescript
const asyncDouble = async (val: number): Promise<number> => val * 2;

const { results } = await PromisePool.withConcurrency(2)
  .for([1, 2, 3])
  .process(async (i) => {
    const result = await asyncDouble(i);

    return result;
  });
```

I would expect `results` to be typed as `number[]`, it's getting typed as `Promise<number>[]` instead. Is this expected, or have I done something wrong?

**Comment by marcuspoehls** [comment]
@calvinwyoung Hey Calvin, thank you for your kind words.

You’re right, the resulting types seem to be off from the expected results. I’ll have a look at what goes wrong here. If you already looked at the code and found the issue, please open a pull request 😃

**Comment by marcuspoehls** [comment]
@calvinwyoung I found the issue. Missed the (very likely happening) promise return type on the async processing function. TypeScript then wrongly resolved the types because I said it should resolve it wrong 😄 

Bugfix release coming soon

**Comment by marcuspoehls** [comment]
@calvinwyoung I’ve tagged and released version `1.5.0` to NPM. Thank you for reporting this issue!

**Comment by calvinwyoung** [comment]
@marcuspoehls Wow thanks for getting that fixed so quickly! I just downloaded the the `1.5.0` release and confirm that fixed this. Thanks so much for the great library!

**Comment by marcuspoehls** [comment]
@calvinwyoung Sweet! 😃 Thank you!
